File: Problems/0189_Rotate_Array/Project_Python3/Rotate_Array.py
# coding: utf-8

import os
import sys
import time

class Solution:
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        temp_nums = nums[:]

        for i in range(len(nums)):
            if i + k < len(nums):
                nums[i + k] = temp_nums[i]
            else:
                nums[(i + k) % len(nums)] = temp_nums[i]

def str_to_int_array(flds):
    if len(flds) <= 0:
        return None
    temp = flds.split(",")
    nums = [0]*len(temp)
    for i in range(len(temp)):
        nums[i] = int(temp[i])
    return nums

def main():
    argv = sys.argv
    argc = len(argv)

    if (argc < 2):
        print("Usage: python %s <testdata.txt>" %(argv[0]))
        exit(0)

    if not os.path.exists(argv[1]):
        print("%s not found..." %argv[1])
        exit(0)

    testDataFile = open(argv[1], "r")
    lines = testDataFile.readlines()

    for temp in lines:
        print("argv[1] = %s" %temp)
        loop_main(temp)

def loop_main(temp):
    flds = temp.replace(" ","").replace("[[","").replace("]]","").rstrip().split("],[")
    nums = str_to_int_array(flds[0])
    k = int(flds[1])
    print("nums = %s, k = %s" %(nums, k))

    time0 = time.time()

    sl = Solution()
    sl.rotate(nums, k)
    print("nums(result) = %s" %nums)

    time1 = time.time()

    print("Execute time ... : %f[s]\n" %(time1 - time0))

if __name__ == "__main__":
    main()

File: Problems/0100_0199/0189_Rotate_Array/Project_CS/Rotate_Array.cs
using System;
using System.Collections.Generic;

public class Solution {
    public void Rotate(int[] nums, int k) {
        int[] temp_nums = new int[nums.Length];
        Array.Copy(nums, temp_nums, nums.Length);
        
        for (int i = 0; i < nums.Length; ++i) {
            if (i + k < nums.Length)
                nums[i + k] = temp_nums[i];
            else
                nums[(i + k) % nums.Length] = temp_nums[i];
        }
    }

    public int[] str_to_int_array(string s)
    {
        if (s.Length <= 0)
            return null;

        string[] flds = s.Split(',');
        int[] nums = new int[flds.Length];

        if (flds.Length <= 0)
            return nums;

        for (int i = 0; i < nums.Length; ++i)
        {
            nums[i] = int.Parse(flds[i]);
        }

        return nums;
    }

    public string output_int_array(int[] nums)
    {
        if (nums.Length <= 0)
            return "";

        string resultStr = "[" +  nums[0].ToString();
 
        for (int i = 1; i < nums.Length; ++i)
        {
            resultStr += "," + nums[i].ToString();
        }

        return resultStr + "]";
    }

    public void Main(string args)
    {
        string[] flds = args.Replace(" ","").Replace("[[","").Replace("]]","").Trim().Split("],[", StringSplitOptions.None);
        int[] nums = str_to_int_array(flds[0]);
        int k = int.Parse(flds[1]);

        Console.WriteLine("nus = " + output_int_array(nums) + ", k = " + k);

        System.Diagnostics.Stopwatch sw = new System.Diagnostics.Stopwatch();
        sw.Start();

        Rotate(nums, k);

        sw.Stop();
        Console.WriteLine("nums(result) = " + output_int_array(nums));
        Console.WriteLine("Execute time ... " + sw.ElapsedMilliseconds.ToString() + "ms\n");
    }
}
